code/DDSCodeTester.cpp

```diff
@@ -1610,6 +1610,199 @@ void dds_topic_examples()
+        if (nullptr == filter_topic_with_parameters)
+        {
+            // Error
+            return;
+        }
+
+        // The ContentFilteredTopic instances can then be used to create DataReader objects.
+        Subscriber* subscriber =
+                participant->create_subscriber(SUBSCRIBER_QOS_DEFAULT);
+        if (nullptr == subscriber)
+        {
+            // Error
+            return;
+        }
+        
+        DataReader* reader_on_filter = subscriber->create_datareader(filter_topic, DATAREADER_QOS_DEFAULT);
+        if (nullptr == reader_on_filter)
+        {
+            // Error
+            return;
+        }
+        
+        DataReader* reader_on_filter_with_parameters =
+                subscriber->create_datareader(filter_topic_with_parameters, DATAREADER_QOS_DEFAULT);
+        if (nullptr == reader_on_filter_with_parameters)
+        {
+            // Error
+            return;
+        }
+        //!--
+    }
+
+    {
+        //DDS_UPDATE_CONTENT_FILTERED_TOPIC
+        
+        // This lambda prints all the information of a ContentFilteredTopic
+        auto print_filter_info = [](
+                const ContentFilteredTopic* filter_topic)
+        {
+            std::cout << "ContentFilteredTopic info for '" << filter_topic->get_name() << "':" << std::endl;
+            std::cout << "  - Related Topic: " << filter_topic->get_related_topic()->get_name() << std::endl;
+            std::cout << "  - Expression:    " << filter_topic->get_filter_expression() << std::endl;
+            std::cout << "  - Parameters:" << std::endl;
+            
+            std::vector<std::string> parameters;
+            filter_topic->get_expression_parameters(parameters);
+            size_t i = 0;
+            for (const std::string& parameter : parameters)
+            {
+                std::cout << "    " << i++ << ": " << parameter << std::endl;
+            }
+        };
+        
+        // Create a DomainParticipant in the desired domain
+        DomainParticipant* participant =
+                DomainParticipantFactory::get_instance()->create_participant(0, PARTICIPANT_QOS_DEFAULT);
+        if (nullptr == participant)
+        {
+            // Error
+            return;
+        }
+
+        // Create a Topic
+        Topic* topic =
+                participant->create_topic("HelloWorldTopic", "HelloWorldTopic", TOPIC_QOS_DEFAULT);
+        if (nullptr == topic)
+        {
+            // Error
+            return;
+        }
+
+        // Create a ContentFilteredTopic
+        ContentFilteredTopic* filter_topic =
+                participant->create_contentfilteredtopic("HelloWorldFilteredTopic", topic, "index > 10", {});
+        if (nullptr == filter_topic)
+        {
+            // Error
+            return;
+        }
+        
+        // Print the information
+        print_filter_info(filter_topic);
```

Comment:
Maybe an expected output can be included either in another snippet or as a comment so the user can check that it is as expected.

Reply:
We talked aboit this large comment and decided it will not contribute much.